channeldb/payments.go

```diff
@@ -1301,7 +1305,23 @@ func deserializeHop(r io.Reader) (*route.Hop, error) {
 		}
 	}
 
-	// If the metatdata type is present, remove it from the tlv map and
+	ampType := uint64(record.AMPOnionType)
+	if ampBytes, ok := tlvMap[ampType]; ok {
+		delete(tlvMap, ampType)
+
+		var (
+			amp    = &record.AMP{}
+			ampRec = amp.Record()
+			r      = bytes.NewReader(ampBytes)
+		)
+		err := ampRec.Decode(r, uint64(len(ampBytes)))
+		if err != nil {
+			return nil, err
+		}
+		h.AMP = amp
```

Comment:
We do have the itest `testSendPaymentAMP` to check the amp-related behavior, but it's weird that the test was already passing - I guess this field is not used in pathfinding, instead, we use `PackHopPayload` when creating amp related hops.

Reply:
yeah exactly, so in test `testSendPaymentAMP`, we already do a minimal check on htlc attempts,
https://github.com/lightningnetwork/lnd/blob/ba4021cad98887b2a63be6b61bf0ad46b07c221a/itest/lnd_amp_test.go#L409-L412

What I'm suggesting is to add checks similar to the invoice HTLCs,
https://github.com/lightningnetwork/lnd/blob/ba4021cad98887b2a63be6b61bf0ad46b07c221a/itest/lnd_amp_test.go#L434-L438

This way we can be sure this field is populated.